### PR TITLE
DOC: `optimize.differential_evolution`: change convergence to reflect implementation

### PR DESCRIPTION
**Pull request by voibit**
Changed documentation to reflect implementation. 

#### Reference issue
-

#### What does this implement/fix?
In the documentation, it is unclear how the differential evolution algorithm converge, if it is based on the standard deviation of the population parameter, or population energies. Changed to be more explicit.   

#### Additional information
-

### CODE REVIEW

**Comment by melissawm** [comment]
Hi @voibit - as a note, for next time, you can use [docs only] on your commit message and save some CI time 😄 see https://scipy.github.io/devdocs/dev/contributor/continuous_integration.html#skipping

As for content, the change looks reasonable but I'm not familiar with these terms so will approve and wait for another review. I will note that this term doesn't get mentioned in the docstring at all so it may need a reference. Thanks!

**Review by melissawm — APPROVED** [review]

**Comment by voibit** [comment]
Thanks @melissawm, Sorry about that, I will read the docs more carefully next time ☺️

**Pull request merged**
